### PR TITLE
UNR-4221 fix: worker checks file ext

### PR DESCRIPTION
**Pull request by thomas-yung**
#### Description
The worker configuration file watcher checks the file extension and only updates if files with the `.worker.json` extension are changed.

#### Tests
Run UE4 and change the files in `UnrealGDKExampleProject\spatial\workers\`

image

image

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Breaking changes:
 - The condition for sending Spatial position updates has been changed, the two variables `PositionUpdateFrequency` and `PositionDistanceThreshold` have now been removed from the GDK settings. To update your project:
   1. Set the value of `PositionUpdateLowerThresholdCentimeters` to the value of `PositionDistanceThreshold` and  the value of `PositionUpdateLowerThresholdSeconds` to 60*(1/`PositionUpdateFrequency`). This will ensure that Actors send Spatial position updates as often as they did before this change.
-  2. Set the value of `PositionUpdateThresholdMaxCentimeters` and `PositionUpdateThresholdMaxSeconds` to larger values than the lower thresholds. 
+  2. Set the value of `PositionUpdateThresholdMaxCentimeters` and `PositionUpdateThresholdMaxSeconds` to larger values than the lower thresholds.
```
Comment (improbable-danny) [inline CHANGELOG.md:15]:
```suggestion
  2. Set the value of `PositionUpdateThresholdMaxCentimeters` and `PositionUpdateThresholdMaxSeconds` to larger values than the lower thresholds. 
```

Comment (improbable-danny) [reply CHANGELOG.md:15]: Is this change intentional?

File: CHANGELOG.md [hunk 2]
```diff
@@ -63,8 +63,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - By default, only an Actor's replicated owner hierarchy will be used when determining which worker should have authority over an actor. Non-replicated Actors are now ignored.
 - Fixed a crash that would sometimes occur when connection to SpatialOS fails.
 - Fixed a crash that occurred when an actor subobject became invalid after applying initial component data.
-- Non-replicated Actors net roles are not touched during startup. 
+- Non-replicated Actors net roles are not touched during startup.
 - Fixed a bug which dropped component updates on authority delegation.
+- Worker configuration watcher only updates when `*.worker.json` files are changed.
```
Comment (improbable-danny) [inline CHANGELOG.md:68]:
```suggestion
- Worker configuration watcher only rebuilds worker configs when `*.worker.json` files are changed.
```
Maybe something like this?

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 3]
```diff
@@ -123,9 +123,17 @@ void FLocalDeploymentManager::StartUpWorkerConfigDirectoryWatcher()
 
 void FLocalDeploymentManager::OnWorkerConfigDirectoryChanged(const TArray<FFileChangeData>& FileChanges)
 {
-	UE_LOG(LogSpatialDeploymentManager, Log,
-		   TEXT("Worker config files updated. Regenerating worker descriptors ('spatial worker build build-config')."));
-	WorkerBuildConfigAsync();
+	for (auto& FileChange : FileChanges)
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp:126]: nit 
```suggestion
	for (const auto& FileChange : FileChanges)
```
Interesting a TArray returns non-const iterators

**Review by improbable-danny — APPROVED** [review]
Looks good!

File: CHANGELOG.md (hunk 1 above)
Comment (thomas-yung) [reply CHANGELOG.md:15]: Nope, I think my IDE did that one 😅

File: CHANGELOG.md [hunk 4]
```diff
@@ -63,8 +63,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - By default, only an Actor's replicated owner hierarchy will be used when determining which worker should have authority over an actor. Non-replicated Actors are now ignored.
 - Fixed a crash that would sometimes occur when connection to SpatialOS fails.
 - Fixed a crash that occurred when an actor subobject became invalid after applying initial component data.
-- Non-replicated Actors net roles are not touched during startup. 
+- Non-replicated Actors net roles are not touched during startup.
```
Comment (improbable-danny) [inline CHANGELOG.md:66]: Fair enough :D 
```suggestion
- Non-replicated Actors net roles are not touched during startup. 
```

File: SpatialGDK/Source/SpatialGDK/Private/SpatialView/ViewDelta.cpp [hunk 5]
```diff
@@ -57,10 +57,10 @@ void ViewDelta::Project(FSubViewDelta& SubDelta, const TArray<Worker_EntityId>&
 
 		// Find the intersection between complete entities and the entity IDs in the view delta, add them to this
 		// delta.
-		if (CompleteIt && DeltaIt && CompleteId == CurrentEntityId && DeltaId == CurrentEntityId)
+		if (CompleteId == CurrentEntityId && DeltaId == CurrentEntityId)
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialView/ViewDelta.cpp:60]: Bad merge?

Comment (thomas-yung) [reply SpatialGDK/Source/SpatialGDK/Private/SpatialView/ViewDelta.cpp:60]: Oh wow, good spot

File: SpatialGDK/Source/SpatialGDK/Private/SpatialView/ViewDelta.cpp [hunk 6]
```diff
@@ -57,10 +57,10 @@ void ViewDelta::Project(FSubViewDelta& SubDelta, const TArray<Worker_EntityId>&
 
 		// Find the intersection between complete entities and the entity IDs in the view delta, add them to this
 		// delta.
-		if (CompleteId == CurrentEntityId && DeltaId == CurrentEntityId)
+		if (CompleteIt && DeltaIt && CompleteId == CurrentEntityId && DeltaId == CurrentEntityId)
```
Comment (jessicafalk) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialView/ViewDelta.cpp:60]: bad merge?

Comment (thomas-yung) [reply SpatialGDK/Source/SpatialGDK/Private/SpatialView/ViewDelta.cpp:60]: ugh again 🤦 rebasing is not liking this one

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 7]
```diff
@@ -123,9 +123,17 @@ void FLocalDeploymentManager::StartUpWorkerConfigDirectoryWatcher()
 
 void FLocalDeploymentManager::OnWorkerConfigDirectoryChanged(const TArray<FFileChangeData>& FileChanges)
 {
-	UE_LOG(LogSpatialDeploymentManager, Log,
-		   TEXT("Worker config files updated. Regenerating worker descriptors ('spatial worker build build-config')."));
-	WorkerBuildConfigAsync();
+	for (const auto& FileChange : FileChanges)
+	{
+		if (FileChange.Filename.EndsWith(".worker.json"))
```
Comment (jessicafalk) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp:128]: nit: I think TArray also has a `ContainsByPredicate` which we could use here as well

**Review by jessicafalk — APPROVED** [review]
one small nit, but overall LGTM!

**Review by jessicafalk — APPROVED** [review]

**Pull request merged**